src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -100,32 +103,53 @@ public void configurePipeline(PipelineConfigurer configurer) {
       stageConfigurer.setOutputSchema(configuredSchema);
       return;
     }
+    try {
+      Schema schema = getSchema(collector);
+      validatePartitionProperties(collector);
 
-    Schema schema = getSchema(collector);
-    validatePartitionProperties(collector);
+      if (configuredSchema == null) {
+        stageConfigurer.setOutputSchema(schema);
+        return;
+      }
 
-    if (configuredSchema == null) {
-      stageConfigurer.setOutputSchema(schema);
-      return;
+      validateConfiguredSchema(configuredSchema, collector);
+      stageConfigurer.setOutputSchema(configuredSchema);
+    } catch (IllegalArgumentException | InvalidStageException e) {
```

Comment:
the code is now really difficult to understand because you can't tell when an exception is thrown, when the collector is used, etc. without tracing through a bunch of code.

The decision about whether to log a warning and move on should be pushed down to the place where the exception originates, not bubbled all the way up here. There is no context here which makes it very easy to do the wrong thing.

I think it would be cleaner for these validate methods to always use the collector and never throw an exception. For validations that try to connect to an external system, those methods can take a boolean that tells them whether to warn and move on, or whether to add an error to the collector.

Reply:
Understood, fixed